File: atcoder/dp_a_frog_1_.hs
-- vicfred
-- https://atcoder.jp/contests/dp/tasks/dp_a
-- dynamic programming
oneDiff :: [Int] -> [Int]
oneDiff h = [abs (i - j) | (i, j) <- zip (tail h) h]

twoDiff :: [Int] -> [Int]
twoDiff h = [abs (i - j) | (i, j) <- zip (tail (tail h)) h]

frog :: [Int] -> [Int]
frog h = dp
  where
    od = oneDiff h
    td = twoDiff h
    dp = 0 : head od : [min (i + j) (k + l) | ((i, j), (k, l)) <- zip (tail $ zip dp od) (zip dp td)]

main :: IO ()
main = do
  n <- read <$> getLine :: IO Int
  h <- map read . words <$> getLine :: IO [Int]
  print . last $ frog h

File: atcoder/dp_a_frog_1.py
# vicfred
# https://atcoder.jp/contests/dp/tasks/dp_a
# dynamic programming
n = int(input())
h = [0]+list(map(int, input().split()))

dp = [0]*(n + 1)
dp[1] = 0
dp[2] = abs(h[2] - h[1])

dp = [0]*(n+1)
for i in range(3, n + 1):
    dp[i] = min(dp[i - 1] + abs(h[i] - h[i - 1]),
                dp[i - 2] + abs(h[i] - h[i - 2])
            )
print(dp[n])

